File: common/src/main/java/dev/number6/message/ChannelMessagesNotificationRequestHandler.java
package dev.number6.message;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.events.SNSEvent;
import com.google.gson.Gson;

public class ChannelMessagesNotificationRequestHandler implements RequestHandler<SNSEvent, String> {

    private final Gson gson = new Gson();
    private final ChannelMessagesHandler channelMessagesHandler;

    public ChannelMessagesNotificationRequestHandler(ChannelMessagesHandler channelMessagesHandler) {
        this.channelMessagesHandler = channelMessagesHandler;
    }

    @Override
    public String handleRequest(SNSEvent event, Context context) {
        LambdaLogger logger = context.getLogger();

        logger.log("Starting SnsMessage Entity Comprehension.");
        logger.log("Received event: " + event);
        logger.log("containing " + event.getRecords().size() + " records.");

        ChannelMessages channelMessages = gson.fromJson(event.getRecords().get(0).getSNS().getMessage(), ChannelMessages.class);

        channelMessagesHandler.handle(channelMessages);
        return "ok";
    }
}

File: common/src/main/java/dev/number6/message/ChannelMessagesNotificationRequestHandler.kt
package dev.number6.message

import com.amazonaws.services.lambda.runtime.Context
import com.amazonaws.services.lambda.runtime.RequestHandler
import com.amazonaws.services.lambda.runtime.events.SNSEvent
import com.google.gson.Gson

open class ChannelMessagesNotificationRequestHandler(private val channelMessagesHandler: ChannelMessagesHandler) : RequestHandler<SNSEvent, String> {
    private val gson = Gson()
    override fun handleRequest(event: SNSEvent, context: Context): String {
        val logger = context.logger
        logger.log("Starting SnsMessage Entity Comprehension.")
        logger.log("Received event: $event")
        logger.log("containing " + event.records.size + " records.")
        val channelMessages = gson.fromJson(event.records[0].sns.message, ChannelMessages::class.java)
        channelMessagesHandler.handle(channelMessages)
        return "ok"
    }

}